src/permission/net_permission.cc

```diff
@@ -0,0 +1,172 @@
+#include "net_permission.h"
+#include "debug_utils-inl.h"
+#include "json_utils.h"
+
+#include <cctype>
+#include <string>
+#include <string_view>
+#include <vector>
+
+namespace node {
+
+namespace permission {
+
+bool ConvertIPToBitset(std::bitset<IPV6_BIT_LEN>* bitset,
+                       const std::string& str,
+                       int ip_version) {
+  int family = ip_version == 6 ? AF_INET6 : AF_INET;
+  int char_len = ip_version == 6 ? IPV6_CHAR_LEN : IPV4_CHAR_LEN;
+  int position = 0;
+  char buf[IPV6_CHAR_LEN];
+  // IP(like 127.0.0.1) in buf is [01111111][00000000][00000000][00000001]
+  if (uv_inet_pton(family, str.c_str(), &buf) == 0) {
+    // Handle char by char
+    for (int i = char_len - 1; i >= 0; i--) {
+      // Handle byte by byte
+      for (int j = 0; j < 8; j++) {
+        bitset->set(position++, !!(buf[i] & (1 << j)));
+      }
+    }
+    return true;
+  }
+  return false;
+}
+
+void NetPermission::Apply(Environment* env,
+                          const std::vector<std::string>& allow,
+                          PermissionScope scope) {
+  // For Debug
+  auto cleanup = OnScopeLeave([&]() { Print(); });
+  using std::string_view_literals::operator""sv;
+  for (const std::string& res : allow) {
+    // Allow multiple item in one flag, like --allow-net-udp=127.0.0.1,localhost
+    const std::vector<std::string_view> addresses = SplitString(res, ","sv);
+    for (const auto& address : addresses) {
+      switch (scope) {
```

Comment:
We don't need this switch. We can just keep it as a simple if statement to simplify the code here.